sc/social/like/plugins/pinterest/browser.py

```diff
@@ -1,80 +1,58 @@
 # -*- coding:utf-8 -*-
+"""Helper view to generate Pinterest widget.
+
+More information:
+* https://developers.pinterest.com/docs/widgets/save/
+* https://developers.pinterest.com/docs/rich-pins/reference/
+"""
+from Acquisition import aq_inner
 from plone import api
-from plone.api.exc import InvalidParameterError
 from Products.Five import BrowserView
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from sc.social.like.interfaces import ISocialLikeSettings
 from sc.social.like.utils import get_content_image
-from sc.social.like.utils import get_language
 from urllib import urlencode
-from zope.component import getMultiAdapter
-
-
-BASE_URL = '//pinterest.com/pin/create/button/'
-PARAMS = '?url={0}&media={1}&description={2}'
 
 
 class PluginView(BrowserView):
+    """Helper view to generate Pinterest widget."""
 
-    pinterest_enabled = False
-    language = 'en'
-
-    metadata = ViewPageTemplateFile('templates/metadata.pt')
+    metadata = None
     plugin = ViewPageTemplateFile('templates/plugin.pt')
     link = ViewPageTemplateFile('templates/link.pt')
 
     def __init__(self, context, request):
-        self.context = context
+        self.context = aq_inner(context)
         self.request = request
-        # FIXME: the following could rise unexpected exceptions
-        #        move it to a new setup() method
-        #        see: http://docs.plone.org/develop/plone/views/browserviews.html#creating-a-view
-        self.portal_state = getMultiAdapter((self.context, self.request),
-                                            name=u'plone_portal_state')
-        self.portal = self.portal_state.portal()
-        self.site_url = self.portal_state.portal_url()
-        self.portal_title = self.portal_state.portal_title()
-        self.url = context.absolute_url()
-        self.image = get_content_image(context, scale='large')
-        self.language = get_language(context)
 
-    def share_url(self):
-        template = BASE_URL + PARAMS
-        return template.format(
-            self.url,
-            self.image_url(),
-            self.context.Title(),
-        )
+    def portal_url(self):
+        portal = api.portal.get()
+        return portal.absolute_url()
 
+    # TODO: check if Pinterest can use Open Graph instead
```

Comment:
Actually Pinterest uses Open Graph;  This method is used just when DnT is activated, the image url needs to be sent as parameter to Pinterest.

Reply:
ok, I'm going to make some more changes according to that.